src/__itests__/App.browser.test.tsx

```diff
@@ -3,18 +3,40 @@ import puppeteer from 'puppeteer';
 let testBrowser: puppeteer.Browser;
 let page: puppeteer.Page;
 
+const host = 'localhost';
+const port = 5003;
+const username = 'admin';
+const password = 'admin';
+
+let getUrl = () => {
+  return 'http://' + host + ':' + port;
+};
+
 jest.setTimeout(30000);
 
 describe('browser_smoke', () => {
   beforeAll(async () => {
+    sessionStorage.setItem('user', 'admin');
     testBrowser = await puppeteer.launch({
       headless: true,
       slowMo: 300,
       args: ['--no-sandbox', '--disable-setuid-sandbox']
     });
     page = await testBrowser.newPage();
     await page.setViewport({ width: 1440, height: 900 });
-    await page.goto('http://admin:admin@localhost:5003/console/service-graph/bookinfo', { waitUntil: 'networkidle2' });
+    await page.goto(getUrl(), { waitUntil: 'load' });
```

Comment:
Just curiosity. How do these async tests work ? They wait until the condition is matched for a timeout ?.

Reply:
you can set the reserved word await or not, If you set wait the code wait until the page go to URL in this case

Reply:
HEre is the documentation about https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md so I added aditional requirement that is wait until load the full page, (this is { waitUntil: 'load' }) because I need the form rendered for fill the inputs with the username and password